jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StaticContextHandler.java

```diff
@@ -0,0 +1,104 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server.handler;
+
+import java.net.URI;
+
+import org.eclipse.jetty.server.Deployable;
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.util.resource.Resource;
+import org.eclipse.jetty.util.resource.Resources;
+
+/**
+ * A ContextHandler that serves Static Content only.
+ *
+ * <p>
+ * The Base Resource represents the static directory root.
+ * </p>
+ */
+public class StaticContextHandler extends ContextHandler implements Deployable
+{
+    /**
+     * Create a StaticContextHandler.
+     */
+    public StaticContextHandler()
+    {
+        super();
+    }
+
+    /**
+     * Create a StaticContextHandler on a specific contextPath.
+     *
+     * @param contextPath the context path to serve static content from
+     */
+    public StaticContextHandler(String contextPath)
+    {
+        this(contextPath, null);
+    }
+
+    /**
+     * Create a StaticContextHandler on a specific contextPath using a configured ResourceHandler.
+     *
+     * @param contextPath the context path
+     * @param resourceHandler the resource handler
+     */
+    public StaticContextHandler(String contextPath, ResourceHandler resourceHandler)
+    {
+        super();
+        setContextPath(contextPath);
+        if (resourceHandler != null)
+            setHandler(resourceHandler);
+    }
+
+    private boolean isResourceHandlerAlreadyPresent(Resource staticDir)
+    {
+        for (Handler handler : getHandlers())
+        {
+            if (handler instanceof ResourceHandler resourceHandler)
+            {
+                Resource baseResource = resourceHandler.getBaseResource();
+                if (baseResource != null)
+                {
+                    URI baseResourceURI = baseResource.getURI();
+                    if (baseResourceURI.equals(staticDir.getURI()))
+                    {
+                        return true;
+                    }
+                }
+            }
+        }
```

Comment:
I still think this is wrong.   If we had `ContextHandler->GzipHandler->ResourceHandler` this method would return false, when the result should be true.   Getting a contained bean is how we have always done this before.